packages/arb-token-bridge-ui/src/hooks/useBalance.ts

```diff
@@ -57,9 +59,18 @@ const useBalance = ({ provider, walletAddress }: UseBalanceProps) => {
         return null
       }
 
-      return [walletAddressLowercased, chainId, 'balance', type] as const
+      return [
+        walletAddressLowercased,
+        chainId,
+        'balance',
+        type,
+        // include l1 and l2 in the query key to cover an edge case
+        // where Arbitrum changes from being L2 to L1, or from L2 to L1
+        l1.network.id,
+        l2.network.id
```

Comment:
Hm, I don't really understand this change. Feels like this hook shouldn't be touched and maybe this needs to be handled outside?